### PR TITLE
add unit test

### PR DESCRIPTION
**Pull request by gaoteng-git**
1. Test profiler parsing and processing. Each test case covers one feature or boundary case.
2. Test range related functions.

### CODE REVIEW

File: tb_plugin/test/test_module_parser.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import math
+import pickle
+import unittest
+
+from tensorboard_plugin_torch_profiler.profiler.data import RunProfileData
+from tensorboard_plugin_torch_profiler.profiler.module_parser import ModuleParser
+
+OP_LIST = "./data/data_op_list_groupby_name.pkl"
+OP_LIST_BY_INPUT = "./data/data_op_list_groupby_name_input.pkl"
```
Comment (nbcsm) [inline tb_plugin/test/test_module_parser.py:9]: hmm, seems not easy to maintain many pickle files.

**Review by gdankel — COMMENTED** [review]
Do you want to run this as part of CI? I'll look into enabling it for this repo.

**Comment by gaoteng-git** [comment]
> Do you want to run this as part of CI? I'll look into enabling it for this repo.

Yes. Not urgent. Thanks a lot!

**Comment by gdankel** [comment]
A few questions, out of curiosity:
1. What is the plan for this going forward? Add additional trace files and resulting baseline files as needed?
2. Isn't the pre-computed baseline files very sensitive to code changes? Will they have to be updated frequently?
3. Will there be instructions for how to re-generate traces (e.g. new features added, schema updates) and precomputed results?
4. I assume the input trace files will have a lot of repetition that doesn't add anything to the test. On the other hand I assume it's easier to maintain this way. But checking in MBs of trace files might not be a good long-term solution. Any thoughts on this?

**Comment by gaoteng-git** [comment]
> A few questions, out of curiosity:
> 
> 1. What is the plan for this going forward? Add additional trace files and resulting baseline files as needed?
> 2. Isn't the pre-computed baseline files very sensitive to code changes? Will they have to be updated frequently?
> 3. Will there be instructions for how to re-generate traces (e.g. new features added, schema updates) and precomputed results?
> 4. I assume the input trace files will have a lot of repetition that doesn't add anything to the test. On the other hand I assume it's easier to maintain this way. But checking in MBs of trace files might not be a good long-term solution. Any thoughts on this?

Good questions. These real model profiling cases are more "real world", so we think they could be more robust to find out code bugs.  Their update frequency may not be too high because the data tables have slow evolving rate. We have re-generate scripts in private repo.
Yes, they are too big in size and hard to maintain in github. We'll make them as internal tests, and manually design some test cases with small input data to submit to this github.

File: tb_plugin/test/test_profiler.py [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+            "ph": "X", "cat": "Kernel", 
+            "name": "void gemmSN_TN_kernel_64addr", "pid": 0, "tid": "stream 7",
+            "ts": 220, "dur": 8,
+            "args": {"correlation": 335, "external id": 0}
+          }]
+        """
+        profile = parse_json_trace(json_content)
+        profile.process()
+        self.assertEqual(profile.op_list_groupby_name[0].device_duration, 0)
+        self.assertEqual(profile.op_list_groupby_name[0].self_device_duration, 0)
```
Comment (guyang3532) [inline tb_plugin/test/test_profiler.py:359]: Should this also test the accumulating device time?

Comment (gaoteng-git) [reply tb_plugin/test/test_profiler.py:359]: In this test case, the Runtime is not invoked by the Operator. So the device durations of this Operator are 0.

Comment (gaoteng-git) [reply tb_plugin/test/test_profiler.py:359]: > Should this also test the accumulating device time?

In this test case, the Runtime is not invoked by the Operator(because external id not equal). So the device durations of this Operator are 0.

File: tb_plugin/test/test_profiler.py [hunk 3]
```diff
@@ -0,0 +1,640 @@
+import json
+import unittest
+
+import tensorboard_plugin_torch_profiler.profiler.trace as trace
```
Comment (nbcsm) [inline tb_plugin/test/test_profiler.py:4]: can use relative import?

**Review by nbcsm — APPROVED** [review]

File: tb_plugin/test/test_profiler.py (hunk 3 above)
Comment (gaoteng-git) [reply tb_plugin/test/test_profiler.py:4]: > can use relative import?

Just follow same style of pytorch tests, such as [test_profiler.py](https://github.com/pytorch/pytorch/blob/master/test/test_profiler.py).
